tst/tests/cast_to_incomplete.un

```diff
@@ -0,0 +1,5 @@
+; ERROR
+
+a : byte
+
+a as void
```

Comment:
We might want to avoid ‘casting to `void`’ to ‘discard’ a value just like in C, but that’s really not that important...
We might want to add a TODO for that tho.

Reply:
When we do implement that, then a test to make sure this isn’t allowed would be:
```
; ERROR

a : @foo
a  as foo ;; or `type foo`, we haven’t decided on the syntax for that yet iirc.
```

Reply:
I see; we could also add in some other way (maybe keyword) to discard a value, mark it unused, etc.

I think updating the test is great, that example is a much better match for an incomplete typecast

Reply:
Discarding a value kind of seems like too narrow of a use case for an entire keyword. I feel like `as void` makes sense in that context because, well, otherwise, it’d have to be invalid anyway, so why not turn it into something useful.

Also, keep in mind that we don’t have order-independent types in the parser atm (and I’m also not sure if `@foo` would be parsed correctly here). I think the parser is still scuffed from when we implemented overloading; at least I don’t think I’ve fixed e.g. checking for duplicate decls yet...

Reply:
> at least I don’t think I’ve fixed e.g. checking for duplicate decls yet...

IIRC I got `redef.un` test to pass (as in it errors as expected)

Reply:
The thing is, just because it errors doesn’t mean the test is working; it could error for a completely unrelated reason. We really have to implement checking for specific error messages in the testing framework...

Reply:
The latter half is a good point. As for the the former half: I specifically fixed `redef.un` by catching redefinition of symbols (reenabling what was commented out, basically).

![image](https://user-images.githubusercontent.com/69637718/215306088-4c0d3960-b9a2-47db-9c09-7653421f5c82.png)

Reply:
I see. And overloads are working? I guess function defs/decls are handled differently, hmm...

Reply:
Sort of. More like I applied my classic bandaid hack technique lmao

![image](https://user-images.githubusercontent.com/69637718/215306234-d5f66913-c6c9-418c-b38d-629e98588c33.png)

Reply:
I see how it is